posting.go

```diff
@@ -493,29 +501,39 @@ func (i *PostingsIterator) nextAtOrAfter(atOrAfter uint64) (segment.Posting, err
 		// rv.freq >= "number of locs", since in a composite field,
 		// some component fields might have their IncludeTermVector
 		// flags disabled while other component fields are enabled
-		if cap(i.nextLocs) >= int(rv.freq) {
-			i.nextLocs = i.nextLocs[0:rv.freq]
-		} else {
-			i.nextLocs = make([]Location, rv.freq, rv.freq*2)
-		}
-		if cap(i.nextSegmentLocs) < int(rv.freq) {
-			i.nextSegmentLocs = make([]segment.Location, rv.freq, rv.freq*2)
+		if rv.freq > 0 {
+			if cap(i.nextLocs) >= int(rv.freq) {
+				i.nextLocs = i.nextLocs[0:rv.freq]
+			} else {
+				i.nextLocs = make([]Location, rv.freq, rv.freq*2)
+			}
+			if cap(i.nextSegmentLocs) < int(rv.freq) {
+				i.nextSegmentLocs = make([]segment.Location, rv.freq, rv.freq*2)
+			}
+			rv.locs = i.nextSegmentLocs[:0]
 		}
-		rv.locs = i.nextSegmentLocs[:0]
 
 		numLocsBytes, err := i.locReader.readUvarint()
 		if err != nil {
 			return nil, fmt.Errorf("error reading location numLocsBytes: %v", err)
 		}
 
 		j := 0
```

Comment:
@sreekanth-cb this last part isn't clear to me.  it seems to only be relating to locations, and not conditional on freq/norm.  can you explain why this is needed?

Reply:
Now that rv.freq can have a zero value, the initializations of those `i.nextLocs` slice won't happen with skip freq/norm configs.
The below code is a workaround to sail through the case where the freq is 0 and we need single allocations of Location{} object for as many location details as we tread further.

line no: 528 nextLoc = &Location{}

Reply:
I see, so in the past, we used freq as a hint for the size to preallocate the number of location structs.  It seems that was always a bit unsafe wasn't it?  It seems the new code is generally safer since we're reading an arbitrary number of locations, the old code was trusting it to be right.

Reply:
yes. that is my thinking here too.